File: 0s-in-factorial/0s-in-factorial.py
def getZeros(n):
    res = 0
    d = 5
    while d <= n:
        res = res + n / d
        d = d * 5
    
    return res

if __name__ == "__main__":
    testcases = [
    (100,24),
    (1000,249),
    (5000,1249),
    ]

    for testcase in testcases:
        num = getZeros(testcase[0])
        if num == testcase[1]:
            print "Currect. In the end of %s!, there are %s 0s." % (testcase[0], num)
        else:
            print "Incurrect. In the end of %s!, there should be %s 0s , result is %s." % (testcase[0], testcase[1], num)

File: 0s-in-factorial/0s-in-factorial.cpp
#include<stdio.h>

int getZeros(int);

int main(int argc, char const *argv[]){
    int testcases[] = {100,1000,5000};
    int results[] = {24,249,1249};

    for (int i = 0; i < 3; ++i){
        int num = getZeros(testcases[i]);
        if (num == results[i]){
            printf("Currect. In the end of %d!, there are %d 0s.\n",testcases[i] ,num);
        }else{
            printf("Incurrect. In the end of %d!, there should be %d 0s , result is %d.",testcases[i],results[i],num);
        }
    }
    return 0;
}

int getZeros(int n){
    int res = 0;
    int d = 5;
    while(d <= n) {
        res += n/d;
        d *= 5;
    }
    return res;
}
